choreolib/src/main/java/choreo/util/ArrayUtil.java

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) Choreo contributors
+
+package choreo.util;
+
+import java.util.function.BiFunction;
+
+public class ArrayUtil {
+  /**
+   * Checks two <code>double</code> arrays for equality with the given function. This returns true
+   * if:
+   * <li>Either both arrays are null, or
+   * <li>Neither is null, the arrays are the same length, and the given function returns true for
+   *     all same-index pairs of elements in the arrays.</>
```

Comment:
I don't believe Javadoc needs a closing brace, but this wouldn't be a valid closing brace anyway.